TheFunctionChangedMyMind/README-en.md

```diff
@@ -0,0 +1,322 @@
+---
+title: The Function that changed my mind
+published: true
+description: How a single function showed me a lot of Kotlin
+tags: kotlin,functions,android,refactor
+cover_image:
+series:
+---
+
+When writting code, we should always have a place for improvement. Improvement can present itself in different ways. A proof of it is how in recent years the attention to CI (_Continuous Integration_) and CD (_Continuous Delivery_) has grown significantly. Same goes for new languages and technologies.
+
+But not all involve changing to a new language, technology or tool. Some of these improvements are simply changing how something works, adding tests, adding abstractions, changing patterns, etc. This is the case for the change I introduced on one of my projects.
+
+**Disclaimer**: If you know me in person or have been in a meetup with me presenting, there's a big chance you already heard me talking about this, even [on YouTube with the GDG Santo Domingo (in spanish)](https://www.youtube.com/watch?v=0KoXxoPHV4o&feature=youtu.be) or Kotlin/Everywhere Guadalajara and Mexico City.
+
+**Another disclaimer**: This article is not about architecture –even when it's a hot topic in the Android community– or best practices or even about how to organize an Android Application. It's just about a set of changes that showed me some Kotlin concepts. Also, I started these changes almost 3 years ago, so by the time this article is published, the libraries from Google –like `core-ktx`– will probably already have something similar or better. And just to be clear: None of the approaches discussed here are right or wrong, all the solutions depend on individual needs and individual reasons to choose them over the others.
```

Comment:
–even though it's a hot topic in the Android community –, best practices or even about how...

Also, I started to implement this aforementioned strategies almost 3 years ago, therefor by the time.... 

will probably being updated or improved.

And just to be clear: It's good to mention that the approaches mentioned in the following lines don't apply to all scenarios nor cases; the solutions mentioned in this article depend on scenarios or needs themselves, besides individual reasons...